classes/DependencyInjection/ContainerProvider.php

```diff
@@ -0,0 +1,90 @@
+<?php
+/**
+ * 2007-2020 PrestaShop and Contributors
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Academic Free License 3.0 (AFL-3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/AFL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2020 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/AFL-3.0 Academic Free License 3.0 (AFL-3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+
+namespace PrestaShop\Module\PrestashopCheckout\DependencyInjection;
+
+use PrestaShop\Module\PrestashopCheckout\Cache\CachePathProviderInterface;
+use PrestaShop\Module\PrestashopCheckout\Configuration\Configuration;
+use Symfony\Component\Config\ConfigCache;
+use Symfony\Component\Config\FileLocator;
+use Symfony\Component\DependencyInjection\ContainerBuilder;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+use Symfony\Component\DependencyInjection\Dumper\PhpDumper;
+use Symfony\Component\DependencyInjection\Loader\YamlFileLoader;
+
+class ContainerProvider
+{
+    /**
+     * @var Configuration
+     */
+    private $configuration;
+
+    /**
+     * @var CachePathProviderInterface
+     */
+    private $cachePathProvider;
+
+    /**
+     * @param Configuration $configuration
+     * @param CachePathProviderInterface $cachePathProvider
+     */
+    public function __construct(Configuration $configuration, CachePathProviderInterface $cachePathProvider)
+    {
+        $this->configuration = $configuration;
+        $this->cachePathProvider = $cachePathProvider;
+    }
+
+    /**
+     * @param string $containerName
+     *
+     * @return ContainerInterface
+     */
+    public function getContainer($containerName)
+    {
+        $containerClassName = ucfirst($this->configuration->getModuleName())
+            . (false === empty($containerName) ? ucfirst($containerName) : '')
```

Comment:
If we always expect a string, I guess `empty($containerName)` is already an empty string?